### ISSUE TITLE
Using transformers disable proxy with additionalProxyRequestHeaders 

### ISSUE THREAD

**Issue by mSenussi** [tomakehurst/wiremock#1143]
- **Overview**: if i use transformers such as  **global-respse-templating** with proxy response contains additionalProxyRequestHeaders. the response gets proxied without additionalProxyRequestHeaders.

- **Environment info**:
    - 1st scenario: 
        - wiremock 2.5.1
        - starting wiremock as a war with custom WarConfiguration to override extensionsOfType and register extensions.
    - 2nd scenario: 
        - wiremock-standalone-2.6.0.jar
        - starting wiremock as: java -jar wiremock-standalone-2.6.0.jar -global-respse-templating.
- **Mapping sample**:
```
{
  "name": "proxy to an endpoint",
  "priority": 2,
  "request": {
    "method": "GET",
    "urlPath": "/api/weather"
  },
  "response": {
    "proxyBaseUrl" : "https://example.com",
    "additionalProxyRequestHeaders": {
      "Subscription-Key": "12345"
    }
  }
}
```

- **Problem**: 
  - without -global-respse-templating the request gets proxied to https://example.com with  "Subscription-Key": "12345".
  - with -global-respse-templating the request gets proxied to https://example.com without  "Subscription-Key": "12345".

- **After debugging**:
  - ResponseTemplateTransformer.transform:67 uses [ResponseDefinitionBuilder.like(responseDefinition)](https://github.com/tomakehurst/wiremock/blob/6176448ba2a051a99160fa88f46f9d8c1779f169/src/main/java/com/github/tomakehurst/wiremock/client/ResponseDefinitionBuilder.java#L52);  which doesn't include additionalProxyRequestHeaders to the copied ResponseDefinition.

**Comment by abartelsmeier** [comment]
Duplicate of #1141

**Comment by tomakehurst** [comment]
Fixed by #1274
